fixed_width/src/de.rs

```diff
@@ -488,29 +485,17 @@ impl<'a, 'de: 'a> serde::Deserializer<'de> for &'a mut Deserializer<'de> {
         self.deserialize_str(visitor)
     }
 
-    fn deserialize_ignored_any<V: Visitor<'de>>(self, visitor: V) -> Result<V::Value, Self::Error> {
-        self.deserialize_any(visitor)
+    // Not supported.
+    fn deserialize_ignored_any<V: Visitor<'de>>(
+        self,
+        _visitor: V,
+    ) -> Result<V::Value, Self::Error> {
+        Err(DeserializeError::WontImplement)
     }
 
-    fn deserialize_any<V: Visitor<'de>>(self, visitor: V) -> Result<V::Value, Self::Error> {
-        let s = self.next_str()?;
-
-        if s.len() == 1 {
-            if s == "1" {
-                visitor.visit_bool(true)
-            } else if s == "0" {
-                visitor.visit_bool(false)
-            } else {
-                let c = s.chars().next().unwrap_or(' ');
-                visitor.visit_char(c)
-            }
-        } else if let Ok(n) = s.parse::<i64>() {
-            visitor.visit_i64(n)
-        } else if let Ok(n) = s.parse::<f64>() {
-            visitor.visit_f64(n)
-        } else {
-            visitor.visit_str(&s)
-        }
+    // FixedWidth is not self describing format should avoid this method.
+    fn deserialize_any<V: Visitor<'de>>(self, _visitor: V) -> Result<V::Value, Self::Error> {
+        Err(DeserializeError::WontImplement)
```

Comment:
I agree this method probably doesn't make a ton of sense for a non self describing format, what do you think about deserializing to a string if the user asks for an Any?

Reply:
There are no ways to infer the type at runtime as I know.
I think most people use derive macro to implement `Deserialize` and `Visitor` trait automatically, and the `Visitor` will forward `Deserializer` to a specific method by the type signature. Also the same story when implementing required traits manually. The format specifications for fixed-width should determine at compile-time rather than at runtime, and it's the only way to go. If the target type signature is `String`, just pass the `Deserializer` to `deserialize_string` but not `deserialize_any`. For more specific usage, people probably should go with field attribute (`#[serde(deserialize_with = "path")]`) to specify their `Deserializer` for that case.

Reply:
Fair enough. FWIW, when I originally wrote this I took inspiration from the serde implementation from csv (which ultimately calls this function): https://github.com/BurntSushi/rust-csv/blob/master/src/deserializer.rs#L206

Reply:
Took some time to figure out rust-csv, I could be wrong, but just leave a note.
The `deserialize_any` is called by `EnumAccess` to determine its variant in `rust-cvs`.
The cvs format is created by hands or machine generated, may contains unexpected values in a column in my experience, so I can understand why the value would pass to `deserialize_any`, but implement global enum maybe a better choice (like [serde_json::value::Value](https://docs.serde.rs/serde_json/value/enum.Value.html)), but also, may be too heavy for that case.
For `fixed_width`, my use case is parsing data generated from fortran fixed format output, its machine generated contents, guarantee typed, and everything should known at compile-time.